lua/nvim-tree/fs.lua

```diff
@@ -317,4 +317,32 @@ function M.print_clipboard()
   return api.nvim_out_write(table.concat(content, '\n')..'\n')
 end
 
+local function copy_to_clipboard(content)
+  vim.fn.setreg('+', content);
+  vim.fn.setreg('"', content);
+  return api.nvim_out_write(string.format('Copy %s to system clipboard! \n', content))
+end
+
+local function path_is_dir(path)
+  local stats = luv.fs_stat(path)
+  return stats and stats.type == 'directory'
+end
+
+function M.copy_filename(node)
+  return copy_to_clipboard(node.name)
+end
+
+function M.copy_path(node)
+  local absolute_path = node.absolute_path
+  local relative_path = utils.path_relative(absolute_path, lib.Tree.cwd)
+  local content = path_is_dir(absolute_path) and utils.path_add_trailing(relative_path) or relative_path
```

Comment:
`path_is_dir` can be replicated by checking `node.entries`, you don't need to call fs_stat and all :)

Reply:
I removed `path_is_dir` function. But by the way, `node.entries` is it a bit unobscure, I can't directly understand what it means.

Reply:
i guess so, but it still is part of this `node` api which is not really well defined :)